node/impl/full/eth.go

```diff
@@ -3,84 +3,70 @@ package full
 import (
 	"context"
 
-	"github.com/ipfs/go-cid"
 	"go.uber.org/fx"
 
-	"github.com/filecoin-project/go-jsonrpc"
 	"github.com/filecoin-project/go-state-types/abi"
 
 	"github.com/filecoin-project/lotus/api"
 	"github.com/filecoin-project/lotus/chain/types/ethtypes"
 	"github.com/filecoin-project/lotus/node/impl/eth"
 )
 
-type EthModuleAPI interface {
-	EthBlockNumber(ctx context.Context) (ethtypes.EthUint64, error)
-	EthAccounts(ctx context.Context) ([]ethtypes.EthAddress, error)
-	EthGetBlockTransactionCountByNumber(ctx context.Context, blkNum ethtypes.EthUint64) (ethtypes.EthUint64, error)
-	EthGetBlockTransactionCountByHash(ctx context.Context, blkHash ethtypes.EthHash) (ethtypes.EthUint64, error)
-	EthGetBlockByHash(ctx context.Context, blkHash ethtypes.EthHash, fullTxInfo bool) (ethtypes.EthBlock, error)
-	EthGetBlockByNumber(ctx context.Context, blkNum string, fullTxInfo bool) (ethtypes.EthBlock, error)
-	EthGetTransactionByHash(ctx context.Context, txHash *ethtypes.EthHash) (*ethtypes.EthTx, error)
+// gatewayWithTrusted exists just to do type assertions on api.Gateway, but we know it won't have
```

Comment:
yes, this is :vomiting_face: but it's part of the DI dance for gateway which is very hard to resolve without it - the "true" gateway API we want to expose is `api.Gateway`, but to make the gateway reusable as a version of all of the modules it's supposed to be able to implement we need to tack these extra pieces on which end up returning "unsupported" when you try to call them